docs/reference/troubleshooting/troubleshooting-unbalanced-cluster.asciidoc

```diff
@@ -0,0 +1,74 @@
+     1                0                 0.0                  260b         260b    47.3gb     43.4gb    100.7gb           46 127.0.0.1 127.0.0.1 CSUXak2 himrst
+--------------------------------------------------
+// TESTRESPONSE[s/\d+(\.\d+)?[tgmk]?b/\\d+(\\.\\d+)?[tgmk]?b/ s/46/\\d+/]
+// TESTRESPONSE[s/CSUXak2 himrst/.+/ non_json]
+
+This response contains the following information that influences balancing:
+
+* `shards` is the current number of shards allocated to the node
+* `shards.undesired` is the number of shards that needs to be moved to other nodes to finish balancing
+* `disk.indices.forecast` is the expected disk usage according to projected shard growth
+* `write_load.forecast` is the projected ingest load associated with this node
+
+A cluster is considered balanced when no shard movements are scheduled (all `shards.undesired` are equal to 0).
+At this state it should have spread workload reasonably. For example, some nodes might have fewer shards but more ingest load.
+
+Some operations such as node restarting, decommissioning, or changing cluster allocation settings
+are disruptive and might require multiple shards to move or rebalance.
+
+Shard movement order is not deterministic and mostly determined by the source and target node readiness to move a shard.
+While rebalancing is in progress some nodes might appear busier then others.
+
+When a shard is allocated to an undesired node it uses the resources of the current node instead of target.
+This might cause a hotspot (disk or CPU) when multiple shards reside on the current node that have not been
+moved to their corresponding targets yet.
+
+If a cluster takes a long time to finish rebalancing you might find the following log entries:
+[source,text]
+--------------------------------------------------
+[WARN][o.e.c.r.a.a.DesiredBalanceReconciler] [10%] of assigned shards (10/100) are not on their desired nodes, which exceeds the warn threshold of [10%]
+--------------------------------------------------
+This is not concerning as long as the number of such shards is decreasing and this warning appears occasionally,
+for example after rolling restarts or changing allocation settings.
+
+If the cluster has this warning for an extended period of time, it is possible that the desired balance has diverged
+too far from the current state. In such cases you can reset the desired balance and calculate a 
+new one that requires less shard movements to get to balanced state. Reset the desired balance 
+using the following API call:
+
+[source,console,id=delete-desired-balance-request-example]
+--------------------------------------------------
+DELETE /_internal/desired_balance
+--------------------------------------------------
+
+If the cluster gets into this state repeatedly, without node restarts or changes in allocation settings,
+you can increase the <<shards-rebalancing-heuristics,`cluster.routing.allocation.balance.threshold`>>.
```

Comment:
Likewise here, no need for "you can".

```suggestion
If the cluster gets into this state repeatedly, without node restarts or changes in allocation settings,
increase the <<shards-rebalancing-heuristics,`cluster.routing.allocation.balance.threshold`>>.
```

Moreover I'd be tempted to suggest this first.